src/inmanta/deploy/scheduler.py

```diff
@@ -54,7 +61,10 @@ def __init__(self) -> None:
         self._update_lock: asyncio.Lock = asyncio.Lock()
 
     async def start(self) -> None:
-        # FIXME[#8009]: read from DB instead
+        resource_mapping, require_mapping = await self.build_resource_mappings_from_db()
+        self._state.construct(resource_mapping, require_mapping)
```

Comment:
I'm surprised new_version doesn't do this?

Reply:
I have the feeling that we need to use the state that was in memory at that time (release of a new version). In the code we do things such as:
```
            deleted_resources: Set[ResourceIdStr] = self._state.resources.keys() - resources_from_db.keys()
```
which gives me the impression that `self._state` is not the most updated one?

Reply:
yes, which is why it is updated with e.g.
 `self._state.update_desired_state(resource, resources_from_db[resource])`

Reply:
Call new version instead of construct